res/controllers/Numark-Mixtrack-Platinum-scripts.js

```diff
@@ -0,0 +1,1415 @@
+        shiftControl: true,
+        shiftOffset: 4,
+        unshift: function() {
+            components.PlayButton.prototype.unshift.call(this);
+            this.type = components.Button.prototype.types.toggle;
+        },
+        shift: function() {
+            this.inKey = 'play_stutter';
+            this.type = components.Button.prototype.types.push;
+        },
+    });
+
+    this.load = new components.Button({
+        inKey: 'LoadSelectedTrack',
+        shift: function() {
+            if (ShiftLoadEjects) {
+                this.inKey = 'eject';
+            }
+            else {
+                this.inKey = 'LoadSelectedTrackAndPlay';
+            }
+        },
+        unshift: function() {
+            this.inKey = 'LoadSelectedTrack';
+        },
+    });
+
+    this.cue_button = new components.CueButton({
+        midi: [0x90 + midi_chan, 0x01],
+        off: 0x01,
+        sendShifted: true,
+        shiftControl: true,
+        shiftOffset: 4,
+        shift: function() {
+            this.inKey = 'start';
+        },
+    });
+
+    this.sync_button = new components.SyncButton({
+        midi: [0x90 + midi_chan, 0x02],
+        off: 0x01,
+        sendShifted: true,
+        shiftControl: true,
+        shiftOffset: 1,
+    });
+
+    this.pfl_button = new components.Button({
+        midi: [0x90 + midi_chan, 0x1B],
+        key: 'pfl',
+        off: 0x01,
+        type: components.Button.prototype.types.toggle,
+        connect: function() {
+            components.Button.prototype.connect.call(this);
+            this.connections[1] = engine.connectControl(this.group, this.outKey, MixtrackPlatinum.pflToggle);
+        },
+    });
+
+    this.hotcue_buttons = new components.ComponentContainer();
+    this.sampler_buttons = new components.ComponentContainer();
+    for (var i = 1; i <= 4; ++i) {
+        this.hotcue_buttons[i] = new components.HotcueButton({
+            midi: [0x94 + midi_chan, 0x18 + i - 1],
+            number: i,
+            sendShifted: true,
+            shiftControl: true,
+            shiftOffset: 8,
+        });
+
+        // sampler buttons 5-8
+        this.sampler_buttons[i] = new components.SamplerButton({
+            midi: [0x94 + midi_chan, 0x18 + i - 1],
+            sendShifted: true,
+            shiftControl: true,
+            shiftOffset: 8,
+            number: i+4,
+            loaded: 0x00,
+            playing: 0x7F,
+        });
+    }
+    this.hotcues = this.hotcue_buttons;
+
+    this.pitch = new components.Pot({
+        inKey: 'rate',
+        invert: true,
+    });
+    if (!this.active) {
+        this.pitch.firstValueReceived = true;
+    }
+
+    var pitch_or_keylock = function (channel, control, value, status, group) {
+        if (this.other.inGetValue() > 0.0 && this.isPress(channel, control, value, status)) {
+            // toggle keylock, both keys pressed
+            engine.setValue(this.group, "keylock", !engine.getValue(this.group, "keylock"));
```

Comment:
You can use the `script.toggleControl` shorthand.
```suggestion
            script.toggleControl(this.group, "keylock");
```